File: src/main/java/be4rjp/sclat/weapon/SPWeapon.java
package be4rjp.sclat.weapon;

import be4rjp.sclat.data.DataMgr;
import be4rjp.sclat.data.PlayerData;
import be4rjp.sclat.manager.SPWeaponMgr;
import org.bukkit.entity.Player;
import org.bukkit.event.EventHandler;
import org.bukkit.event.Listener;
import org.bukkit.event.block.Action;
import org.bukkit.event.player.PlayerInteractEvent;

/**
 * @author Be4rJP
 */
public class SPWeapon implements Listener {
    //スペシャルウエポンのリスナー部分
    @EventHandler
    public void onClickSPWeapon(PlayerInteractEvent event) {
        Player player = event.getPlayer();
        Action action = event.getAction();
        PlayerData data = DataMgr.getPlayerData(player);

        if (player.getInventory().getItemInMainHand() == null || player.getInventory().getItemInMainHand().getItemMeta() == null || player.getInventory().getItemInMainHand().getItemMeta().getDisplayName() == null)
            return;

        if (action.equals(Action.RIGHT_CLICK_AIR) || action.equals(Action.RIGHT_CLICK_BLOCK))
            SPWeaponMgr.UseSPWeapon(player, player.getInventory().getItemInMainHand().getItemMeta().getDisplayName());

    }
}

File: src/main/kotlin/be4rjp/sclat/weapon/SPWeapon.kt
package be4rjp.sclat.weapon

import be4rjp.sclat.data.DataMgr.getPlayerData
import be4rjp.sclat.manager.SPWeaponMgr
import org.bukkit.event.EventHandler
import org.bukkit.event.Listener
import org.bukkit.event.block.Action
import org.bukkit.event.player.PlayerInteractEvent

/**
 *
 * @author Be4rJP
 */
class SPWeapon : Listener {
    // スペシャルウエポンのリスナー部分
    @EventHandler
    fun onClickSPWeapon(event: PlayerInteractEvent) {
        val player = event.getPlayer()
        val action = event.getAction()
        getPlayerData(player)

        if (player.inventory.itemInMainHand
                .itemMeta == null
        ) {
            return
        }

        if (action == Action.RIGHT_CLICK_AIR || action == Action.RIGHT_CLICK_BLOCK) {
            SPWeaponMgr.useSPWeapon(
                player,
                player.inventory.itemInMainHand.itemMeta!!
                    .displayName,
            )
        }
    }
}
